pepper-apis/src/main/java/org/broadinstitute/ddp/filter/StudyLanguageResolutionFilter.java

```diff
@@ -80,6 +80,25 @@ private static Locale getPreferredLocale(Handle handle, String acceptLanguageHea
     }
 
     private static LanguageDto convertLocaleToLanguageDto(Handle handle, Locale preferredLocale) {
-        return LanguageStore.getOrCompute(handle, preferredLocale.getLanguage());
+        String lang = preferredLocale.getLanguage();
+        LanguageDto languageDto = LanguageStore.getOrCompute(handle, lang);
+
+        if (languageDto == null) {
+            languageDto = LanguageStore.getOrCompute(handle, getNewerLanguageCode(lang));
+        }
+        return languageDto;
+    }
+
+    private static String getNewerLanguageCode(String oldCode) {
+        // Java converts some language codes to older versions.  If the language is one of these older codes, try
+        // converting it to the newer version
+        if ("iw".equals(oldCode)) {
+            return "he";
+        } else if ("ji".equals(oldCode)) {
+            return "yi";
+        } else if ("in".equals(oldCode)) {
+            return "id";
```

Comment:
Any way to make these constants? Unlike `en` or `fr`, it's a bit difficult to make out what these are at first glance.

Reply:
Sure.  Good idea.